### PR TITLE
more python

### PR DESCRIPTION
**Pull request by chrisnc**
- add python scripts corresponding to the C examples
- do diff-based tests with python/scons instead of `diff`
- rename shell scripts to reflect their bash-ness
- update github workflow with python3.11, black 22.10.0, enable pip caching

### CODE REVIEW

File: py/scripts/lith-sign [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            f"usage: {sys.argv[0]} <secret-key-file> <message-file> <signature-file>",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+
+    with open(sys.argv[1], "rb") as sk_file:
+        sk = sk_file.read()
+
+    with open(sys.argv[2], "rb") as msg_file:
+        msg = msg_file.read()
```
Comment (xialu000) [inline py/scripts/lith-sign:23]: 1-liner: `msg = Path(sys.argv[2]).read_bytes()`

File: setup.py [hunk 2]
```diff
@@ -34,11 +34,17 @@
...
     keywords=["x25519", "gimli", "signatures"],
-    package_dir={"lithium": "py_src"},
+    package_dir={"lithium": "py"},
     packages=["lithium"],
     include_package_data=True,
     package_data={"build_tools": ["ffibuilder.py"]},
     install_requires=["cffi>=1.12.0"],
     setup_requires=["cffi>=1.12.0", "setuptools_scm"],
     cffi_modules=["ffibuilder.py:ffibuilder"],
+    scripts=[
```
Comment (xialu000) [inline setup.py:44]: nit: I think we should prefer to use `entry_points`; see discussion here: https://stackoverflow.com/questions/18787036/difference-between-entry-points-console-scripts-and-scripts-in-setup-py

Comment (chrisnc) [reply setup.py]: done

**Review by xialu000 — APPROVED** [review]

**Review by xialu000 — DISMISSED** [review]

**Review by elafargue — APPROVED** [review]

File: py/scripts/gimli_hash.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Part of liblithium, under the Apache License v2.0.
```
Comment (sethterashima) [inline py/scripts/gimli_hash.py:1]: Consider adding `#!/usr/bin/env python3` to stand-alone scripts.

**Review by sethterashima — DISMISSED** [review]

File: py/scripts/gimli_hash.py (hunk 3 above)
Comment (chrisnc) [reply py/scripts/gimli_hash.py:1]: They were there when this was a standalone script, but this is using the `setuptools` `entry_points` mechanism to produce the scripts, and this file itself is not a runnable script (and also does not have execute permissions). See @xialu000's comment here: https://github.com/teslamotors/liblithium/pull/46#discussion_r1024702117

Comment (chrisnc) [reply py/scripts/gimli_hash.py]: I've renamed the directory to avoid confusion.

**Review by sethterashima — APPROVED** [review]

**Pull request merged**